docs/paper/dev/api/how-do-plugins-work.md

```diff
@@ -0,0 +1,101 @@
+safe to register event listeners and other resources that the plugin needs to run, however often not safe to interact 
+with a lot of APIs.
+
+This is when you can also open database connections, start threads, and other things that are not safe to do in the
+`onLoad` method.
+
+### Disabling
+
+When a plugin is disabled, its `onDisable` method is called. This method is used to clean up any resources that the 
+plugin has allocated. This method is called before all plugins are unloaded, and is meant for any cleanup that needs to
+be done before the plugin is unloaded. This may include saving data to disk or closing connections to databases.
+
+## Event Listeners
+
+Events are a way for plugins to listen to things that happen in the server. Events are fired when something happens,
+and plugins can listen to these events and run code when they are fired. For example, the `PlayerJoinEvent` is fired 
+when a player joins the server. Plugins can listen to this event and run code when it is fired. This is a more 
+performant way to run code when something happens, as opposed to constantly checking if something has happened.
+
+Some events are cancellable. This means that when the event is fired, it can be cancelled. For example, the
+`PlayerMoveEvent` is cancellable. This means that when it is cancelled, the player will not move. This is useful for 
+things like anti-cheat, where you want to cancel the event if the player is moving too fast.
+
+It is important to think about how "hot" an event is when writing event listeners. A "hot" event is an event that is fired
+very often. For example, the `PlayerMoveEvent` is fired every time a player moves. This means that if you have a lot of
+expensive code in your event listener, it will be run every time a player moves. This can cause a lot of lag. It is
+important to keep event listeners as lightweight as possible.
+
+## Commands
+
+Commands are a way for players to run code on the server. Commands are registered by plugins and can be run by players.
+For example, the `/help` command is registered by the server and can be run by players. Commands can be run by players
+by typing them in the chat or by running them from a command block. Commands can also be run by other plugins.
+
+Commands can have arguments. For example, the `/give` command takes an argument for the player to give the item to and
+an argument for the item to give. Arguments are separated by spaces. For example, the command `/give Notch diamond` will
+give the player named Notch a diamond. Note here that the arguments are `["Notch", "diamond"]`.
+
+### Permissions
+
+Permissions a powerful tool and can be used to control who can run commands and who can listen to events. Permissions 
+are registered by plugins and can be checked by other plugins. Permissions can be granted to players and groups. 
+Permissions can also be granted to other permissions. For example, the `paper.command.help` permission is granted to the 
```

Comment:
can probably be described as a hierarchical structure instead of "permissions being granted other permissions"